pkg/apis/eksctl.io/v1alpha4/defaults.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha4
+
+// SetNodeGroupDefaults will set defaults for a given nodegroup
```

Comment:
This wasn't moved here before due to the concerns regarding `ami` package and its dependencies.

Reply:
Why not keep this in the nodegroup_filter package as private and only expose `ValidateNodeGroupsAndSetDefaults()`? Since this is less abstracted and not really useful by itself, or is it? :/

Reply:
So `NodeGroupFilter` is currently designed for CLI use-cases, hence it lives in `cmduitls`.
This is here so that one can construct `ClusterConfig`, and apply defaults and validation in a deamon/operator context or whatever other context. I'm just trying to draw a clear line between CLI and API.